publish-maven.gradle

```diff
@@ -1,172 +1,152 @@
-apply plugin: 'maven'
+apply plugin: 'maven-publish'
+
+//alias install -> publishToMavenLocal
+task install(dependsOn:'publishToMavenLocal'){}
 
 // exclude hadoop install sub projects
 def hadoopInstalls = ['cdh5', 'hadoop26', 'hdp22', 'phd21', 'phd30']
 if (hadoopInstalls.contains(project.name)) {
-	// Disable installing hadoop distro artifacts
-	install.enabled = false
-	// Disable publishing hadoop distro artifacts
 	project.tasks.findByPath("artifactoryPublish")?.enabled = false
 }
 
-install {
-	repositories.mavenInstaller {
-		customizePom(pom, project)
-	}
-}
 
-def customizePom(pom, gradleProject) {
-	pom.whenConfigured { generatedPom ->
-		// respect 'optional' and 'provided' dependencies
-		gradleProject.configurations.optional.each { dep ->
-			generatedPom.dependencies.find { it.artifactId == dep.name }?.optional = 'true'
-		}
-		gradleProject.configurations.provided.each { dep ->
-			generatedPom.dependencies.find { it.artifactId == dep.name }?.scope = 'provided'
-		}
 
-		// eliminate test-scoped dependencies (no need in maven central poms)
-		generatedPom.dependencies.removeAll { dep ->
-			dep.scope == 'test'
-		}
+// load versions
+def versions = new Properties()
+def propertiesFile = new File('dependencies.properties')
+propertiesFile.withInputStream {
+	versions.load(it)
```

Comment:
I'll take a look if that's possible. I agree that would be better. Also will avoid rework.